src/_pytest/pythonpath.py

```diff
@@ -0,0 +1,28 @@
+import sys
+from typing import List
+
+import pytest
+from pytest import Config
+from pytest import Parser
+
+
+added_paths: List[str]
+
+
+def pytest_addoption(parser: Parser) -> None:
+    parser.addini("pythonpath", type="paths", help="Add paths to sys.path", default=[])
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_load_initial_conftests(early_config: Config) -> None:
+    global added_paths
+    added_paths = [str(p) for p in early_config.getini("pythonpath")]
+    # `pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`
+    for path in reversed(added_paths):
+        sys.path.insert(0, path)
+
+
+def pytest_unconfigure() -> None:
+    global added_paths
+    for path in added_paths:
+        sys.path.remove(path)
```

Comment:
@bluetech Here's the cleanup code. I'm sure glad you recognized that as a potential problem.

Reply:
I think using `sys.path.remove` makes some sense - we all add, then we always remove. At least it's got some symmetry to it. So let's try it and see if anyone ever complains.